arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1936 @@
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/subscriptionId"
+        }, {
+          "name": "resourceGroupName",
+          "in": "path",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/api-version"
+        }],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/IotHubSkuDescriptionListResult"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/eventHubEndpoints/{eventHubEndpointName}/ConsumerGroups": {
+      "get": {
+        "tags": ["GET"],
+        "summary": "Get EventHub Consumer Groups for an IotHub",
+        "description": "Get EventHub Consumer Groups for an IotHub",
+        "operationId": "IotHubResource_ListEventHubConsumerGroups",
+        "consumes": [],
+        "produces": ["application/json", "text/json"],
+        "parameters": [{
+          "name": "resourceName",
+          "in": "path",
+          "description": "Name of the resource.",
+          "required": true,
+          "type": "string"
+        }, {
+          "name": "eventHubEndpointName",
+          "in": "path",
+          "description": "Name of the event hub endpoint.",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/subscriptionId"
+        }, {
+          "name": "resourceGroupName",
+          "in": "path",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/api-version"
+        }],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/StringListResult"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/eventHubEndpoints/{eventHubEndpointName}/ConsumerGroups/{groupName}": {
+      "get": {
+        "tags": ["GET"],
+        "summary": "Get EventHub Consumer Group for an IotHub",
+        "description": "Get EventHub Consumer Group for an IotHub",
+        "operationId": "IotHubResource_GetEventHubConsumerGroup",
```

Comment:
@rkmanda 
Food for thought: 
I think the operationID should be named as "EventHubConsumerGroup_Get", as it is an operation on **EventHubConsumerGroup**. 

Everything will be an operation on the IotHubResource any which ways. Let me give you an example. 

In ARM, the first thing that one needs to do is create a resource group. It is **only after this** you can do something   else in Azure. Does that mean all the operationIds should be named after **ResourceGroup_someaction** ?

Just like resource group the the IotHubResourcename is another path parameter.

When you categorize things logically into different operation groups (the noun before underscore in the operationId) it becomes easier for users to find things using intellisense.
`client.EventHubConsumberGroup.create|get|list|delete`    **v/s** the user getting a big list when he/she does
client.IotHubResource**.**blah1|blah2, etc..